include/oxen/quic/types.hpp

```diff
@@ -8,6 +8,25 @@
 
 namespace oxen::quic
 {
+    class Stream;
+    struct Address;
+
+    namespace concepts
```

Comment:
I'm not a huge fan of the separate namespace for concepts because a "concept" is just a generic language feature: it feels a bit like making a separate namespace dedicated only to any function that uses templates, or putting all your integers inside an `integers` namespace.

But I can live with it if you feel strongly.

Reply:
I'm not married to the idea. It scratched an itch for me organizationally, but I can inline it to make it a bit less gauche